in-canvas-for-x width:auto instead of width:inherit

RvWensen opened the pull request:
Issue fix.
Discussion can be found https://github.com/zurb/foundation-sites/issues/11124

RvWensen committed:
in-canvas-for-x width:auto instead of width:inherit

Issue fix.
Discussion can be found https://github.com/zurb/foundation-sites/issues/11124

ncoden commented:
Hi @RvWensen,

Here are some recommentations your future PRs:
* Use the same "format" as others PRs for your PR and commits titles: `feat:`, `fix:`, `docs:`, `refactor:`, `clean:`, `chore:`...
* Provide a list of the changes mades with their reasons (even if you end in repeating yourself with what was said in the related issue).
* If your PR closes an issue, said it explicitely (like "Closes #XXX" or "Fixes #XXX")

RvWensen commented:
Thanks for the feedback!

ncoden commented on scss/components/_off-canvas.scss:
```diff
@@ -417,7 +417,7 @@ $breakpoint: small
   position: static;
   background: inherit;
-  width: inherit;
+  width: auto;
```
This makes sense for `width: auto`, but @SassNinja was there any reasons for going for `inherit` in the first place for properties that does not inherit their value by default ?

ncoden replied:
poke @SassNinja

ncoden replied:
Done some research. The incanvas feature always used `inherit`, without explicit reasons given. Most font properties should inherit but not `background`, `width`, `overflow`, `transition` as this is not their default value.

SassNinja replied:
@ncoden the in-canvas is supposed to just revert the off-canvas styles. Seems I was overusing the `inherit` a bit ;)
You're replacements of the properties in talk look good to me 👍

ncoden committed:
fix: reset all OffCanvas properties when in in-canvas mode

> Done some research. The incanvas feature always used `inherit`, without explicit reasons given. Most font properties should inherit but not `background`, `width`, `overflow`, `transition` as this is not their default value. I'll replace all of them.

See https://github.com/zurb/foundation-sites/pull/11141#discussion_r180879166

SassNinja commented:
lgtm 👍 

The `inherit` was simply overused. Looks better now!
